config.nix

```diff
@@ -90,14 +98,13 @@
      configureFlags = with ps.stdenv; (drv.configureFlags or []) ++ lib.optional hostPlatform.isWindows "--enable-static --disable-shared";
    });
 
-    haskell = lib.recursiveUpdate ps.haskell {
-      compiler.ghc842 = (ps.haskell.compiler.ghc842.override ghcPkgOverrides).overrideAttrs ghcDrvOverrides;
-      compiler.ghc843 = (ps.haskell.compiler.ghc843.override ghcPkgOverrides).overrideAttrs ghcDrvOverrides;
-      compiler.ghc844 = (ps.haskell.compiler.ghc844.override ghcPkgOverrides).overrideAttrs ghcDrvOverrides;
-      compiler.ghc861 = (ps.haskell.compiler.ghc861.override ghcPkgOverrides).overrideAttrs ghcDrvOverrides;
-      compiler.ghc862 = (ps.haskell.compiler.ghc862.override ghcPkgOverrides).overrideAttrs ghcDrvOverrides;
-      compiler.ghc863 = (ps.haskell.compiler.ghc863.override ghcPkgOverrides).overrideAttrs ghcDrvOverrides;
-      compiler.ghc864 = (ps.haskell.compiler.ghc864.override ghcPkgOverrides).overrideAttrs ghcDrvOverrides;
-    };
+   haskell = lib.recursiveUpdate ps.haskell {
+       compiler = lib.mapAttrs (_name: compiler: (compiler.override ghcPkgOverrides).overrideAttrs ghcDrvOverrides)
+         # these patches (ghcPkgOverrides and ghcDrvOverrides) only apply to vanilla source ghcs. Not ghcjs or binary distributions.
+         # we also ignore ghc82. And are only concerned with ghc84+
+         (lib.filterAttrs
+           (n: _value: !(ps.stdenv.targetPlatform.isGhcjs or false) # we want to apply this only to non-ghcjs ones. As we do some ghc <- ghcjs mapping for ghcjs.
```

Comment:
Maybe use a let binding for this filter predicate so that it can be explained better. like `needsPatches = name: ...`

Reply:
Yeah, this nested map/filter thing is a bit hard to read, some judicious factoring out of functions would help.